app/routes/goal_routes.py

```diff
@@ -0,0 +1,93 @@
+from flask import Blueprint, jsonify, request, make_response, abort
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+
+
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    '''Creates a new Goal record. Title is only required param.'''
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except KeyError:
+        return {"details": "Invalid data"}, 400
+    
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": Goal.goal_response(new_goal)}, 201
+
+
+@goal_bp.route("", methods=["GET"])
+def get_all_goals():
+    '''Returns all Goals. Default sorted by id.'''
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        goals_response.append(Goal.goal_response(goal))
+    return jsonify(goals_response)
+
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    '''Request information about a specific Goal.'''
+    goal = Goal.validate_goal(goal_id)
+    return {"goal": Goal.goal_response(goal)}
```

Comment:
Flask will handle this correctly (jsonify the dictionary and add a code of 200) but I recommend being explicit about the return type.  The clearer the return type is, the easier it is for someone reading the code to understand exactly what's intended here.
```suggestion
    return jsonify({"goal": Goal.goal_response(goal)}), 200
```